### PR TITLE
feat(framework:skip) Update Docker Compose `distributed` files for 1.13.0

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

Updates the Docker Compose complete setup to incorporate the upcoming 1.13.0 changes.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: src/docker/distributed/server/compose.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ services:
   superlink:
     image: flwr/superlink:${FLWR_VERSION:-1.12.0}
```
Comment (chongshenng) [inline src/docker/distributed/server/compose.yml:3]:
```suggestion
    image: flwr/superlink:${FLWR_VERSION:-1.12.0}
```
Same as [this comment](https://github.com/adap/flower/pull/4413#discussion_r1827899886). This should be `nightly` for now, until `1.13` is released.

File: src/docker/distributed/server/compose.yml [hunk 2]
```diff
@@ -17,38 +21,28 @@ services:
...
       - 9092:9092
+      - 9093:9093
 
-  superexec:
+  serverapp:
     build:
       context: ${PROJECT_DIR:-.}
       dockerfile_inline: |
-        FROM flwr/superexec:${FLWR_VERSION:-1.12.0}
+        FROM flwr/serverapp:${FLWR_VERSION:-1.12.0}
```
Comment (chongshenng) [inline src/docker/distributed/server/compose.yml:30]: Same comment as above.

File: src/docker/distributed/server/compose.yml [hunk 3]
```diff
@@ -17,38 +21,28 @@ services:
...
 
         WORKDIR /app
         COPY --chown=app:app pyproject.toml .
         RUN sed -i 's/.*flwr\[simulation\].*//' pyproject.toml \
           && python -m pip install -U --no-cache-dir .
 
-        ENTRYPOINT ["flower-superexec"]
+        ENTRYPOINT ["flower-serverapp"]
```
Comment (chongshenng) [inline src/docker/distributed/server/compose.yml:37]:
```suggestion
        ENTRYPOINT ["flwr-serverapp"]
```

**Review by chongshenng — CHANGES_REQUESTED** [review]
Thanks @Robert-Steiner. Similar to #4413, let's set the image tags to `nightly`.

**Review by chongshenng — APPROVED** [review]
lgtm!

**Review by tanertopal — APPROVED** [review]
Lgtm!

**Pull request merged**